filebeat/_meta/fields.common.yml

```diff
@@ -3,19 +3,13 @@
   description: >
     Contains log file lines.
   fields:
-    - name: source
+    - name: log.source
```

Comment:
If `source` and `offset` is implemented as elements of a `common.MapStr` we should add a `log` group and two elements; `source` and `offset`. It looks weird that in the event it is displayed as `{"log":{ "offset": 4, "source": "myfile"}}` with hierarchical keys, but in the documentation there is a single key which contains a dot (`log.offset` and `log.source`).

Reply:
Did it that way because the "log" object already exists in ECS and didn't want to get a duplicate error.

Now I tried and seems that objects don't cause a duplicate field error. But now I'm curious, what is the difference with other fields that were already specified like this? For example `fileset.module` and `fileset.name`.

Reply:
I looked into how `fileset.module` and `fileset.name` are added to events. It seems to be the same way as you did. So I am willing to let this go. Nevertheless the implementation is inconsistent with the documentation. It's not a big deal, because it's only a small implementation detail which is hidden from the users.

In case of `fileset.module` the key includes a dot. To be consistent with the documentation events should look like this:
```json
{
  "fileset.module": "nginx",
}
```

But in reality `module` is implemented as a submap of `fileset` which leads to the following event:
```
{
  "fileset": {
    "module": "nginx" }
}
```
But in the end it does not make a difference. In the index template everything is correct. So disregard my comment.
Now I am wondering why we need `group` type at all...

Reply:
Historically `a.b` was not possible and we need groups everywhere. We then introduced logic to be able to define `a.b` directly and the scripts do the right things for template, docs generations etc.

Events that end up in Elasticsearch should always have objects as we should use `MapStr.Put(...)` which does it automtically. 

For the docs generation: If there are differences, we should improve the generation script.